manifests/port.pp

```diff
@@ -12,15 +12,16 @@
 #   }
 #
 # @param context A port-context name
-# @param protocol Either tcp or udp. If unset, omits -p flag from semanage.
-# @param port An network port number, like '8514'
-# @param argument An argument for semanage port. Default: "-a"
+# @param protocol Either tcp or udp.
+# @param port An network port number, like '8514', or a range like '1234-4321'
+# @param argument *deprecated*, ignored
 #
 define selinux::port (
   $context,
   $port,
+  $ensure = 'present',
```

Comment:
> I honestly would like to keep the type as a String (or Variant[Integer, String]) to reflect this 
> as it is not immediately obvious that an array represents a range instead of individual
> ports, and it matches semanage output.

Yes indeed the Array way is not so obvious as the String way, but allows to fail faster if somebody tries to pass wrong numbers.

But It can be coded  as you like. Both ways are ok for me.